s3fs/core.py

```diff
@@ -137,16 +139,39 @@ def connect(self, refresh=False):
                 from botocore import UNSIGNED
                 conf = Config(connect_timeout=self.connect_timeout,
                               read_timeout=self.read_timeout,
-                              signature_version=UNSIGNED, **self.kwargs)
-                s3 = boto3.Session().client('s3', config=conf)
+                              signature_version=UNSIGNED)
+                s3 = boto3.Session(**self.kwargs).client('s3', config=conf)
             else:
                 conf = Config(connect_timeout=self.connect_timeout,
                               read_timeout=self.read_timeout)
-                s3 = boto3.Session(self.key, self.secret,
+                s3 = boto3.Session(self.key, self.secret, self.token,
                                    **self.kwargs).client('s3', config=conf)
             self._conn[tok] = s3
         return self._conn[tok]
 
+    def get_delegated_s3pars(self, exp=3600):
+        """Get temporary credentials, apropriate for sending across a network
+
+        Parameters
+        ----------
+        exp : int
+            Time in seconds that credentials are good for
+
+        Returns
+        -------
+        dict of parameters
+        """
+        if self.anon:
+            return {'anon': True}
+        if self.token:  # already has temporary cred
+            return {'key': self.key, 'secret': self.secret, 'token': self.token,
+                    'anon': False}
+        sts = boto3.Session(self.key, self.secret, self.token,
+                            **self.kwargs).client('sts')
+        cred = sts.get_session_token(DurationSeconds=3600)['Credentials']
+        return {'key': cred['AccessKeyId'], 'secret': cred['SecretAccessKey'],
+                'token': cred['SessionToken'], 'anon': False}
```

Comment:
Plausibly, but perhaps we want new temporary credentials every time, and of course they are designed to expire (so we would have to at least check for expiry).